### PR TITLE
Search Component enhancement and Fix for small screens

### PR DESCRIPTION
**Pull request by DivyanshiSingh**
Enhancement: #101 
1. Implemented a configurable search component
2. Brought search component to the horizontal nav in smaller screens
3. Changed functionality of search icon: User can click on the icon to search

s1
s2
s3

### CODE REVIEW

**Comment by DivyanshiSingh** [comment]
@jennydaman have a look at this. I was thinking why are we not using react-router-dom v5? (this will provide us the functionality to switch between routes using useHistory() hooks which is not available in 4.3.1)

File: src/components/Navbar/Navbar.js [hunk 1]
```diff
@@ -35,6 +35,7 @@ const navLinks = [
  * Conditionally renders a list of links into a <Nav>.
  */
 const Navigation = ({ store }) => {
+  const [searchKey, setSearchKey] = useState('');
```
Comment (jennydaman) [inline src/components/Navbar/Navbar.js:38]: yay, hooks!

File: src/components/Navbar/Navbar.js [hunk 2]
```diff
@@ -80,9 +90,6 @@ const StatefulLoginButton = ChrisStore.withStore(LoginButton);
 
 const HeaderTools = (
   <PageHeaderTools>
-    <PageHeaderToolsItem>
```
Comment (jennydaman) [inline src/components/Navbar/Navbar.js:83]: I put `<Search />` in `<PageHeaderToolsItem>` for the sake of using `<PageHeaderToolsItem>`. Perhaps for better "integration" with Patternfly's design opinions. I assume the reason you are changing this is for greater flexibility with placement?

**Comment by jennydaman** [comment]
The screenshots were much appreciated.

Maybe we should collapse the word "Seach" and show the symbolic icon only on small screens.

I haven't looked into react-router-dom v5. I'd say go for it. Aside from Patternfly 3 --> 4, another objective of this issue is to modernize and update all dependencies. It can get messy because incompatibilities happen every time.

File: src/components/Navbar/Navbar.js (hunk 2 above)
Comment (DivyanshiSingh) [reply src/components/Navbar/Navbar.js:83]: Yes the placement was the reason to change the position of Search Component

**Comment by DivyanshiSingh** [comment]
I have changed the react-router-dom version and the Search icon changes you asked. Please have a look.

**Comment by jennydaman** [comment]
I see that you've removed the text. Iconography should speak for itself, good thinking.

screenshot

Once expanded, the icon gets misplaced.

btw, the target branch https://github.com/FNNDSC/ChRIS_store_ui/tree/pf4 itself is a development branch, Changes can be merged into it whenever you feel like, so let me know.

**Comment by DivyanshiSingh** [comment]
> Once expanded, the icon gets misplaced.
I have fixed this, It was a css issue.

**Comment by jennydaman** [comment]
Please create new PRs for subsequent features, such as for `Button`

**Comment by jennydaman** [comment]
Hmm, on wide screens the search bar's left-justify looks weird. Maybe have the bar expand to take up all the space it can have, or restore the right-justified design.

screenshot

**Pull request merged**